sale_start_end_dates/tests/test_sale_start_end_dates.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (C) 2018 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+from odoo.exceptions import ValidationError
+import datetime
+
+
+class TestSaleStartEndDates(TransactionCase):
```

Comment:
switch this to a SavePointCase
As you have a lot of test cases it will speed them up by ~50% as it won't run setup before each one. It will rollback to savepoint instead